src/DoctrineBatchUtils/BatchProcessing/SelectBatchIteratorAggregate.php

```diff
@@ -0,0 +1,138 @@
+<?php
+declare(strict_types=1);
+
+namespace DoctrineBatchUtils\BatchProcessing;
+
+use ArrayIterator;
+use DoctrineBatchUtils\BatchProcessing\Exception\MissingBatchItemException;
+use Doctrine\ORM\AbstractQuery;
+use Doctrine\ORM\EntityManagerInterface;
+use IteratorAggregate;
+use Traversable;
+
+/**
+ * 'Read' focused batching iterator that will issue a clear to the entity manager
+ * every batch size to 'detach' the managed objects and therefore make them entities only for the
+ * purpose of reading without manually attaching them back to the EntityManager.
+ */
+final class SelectBatchIteratorAggregate implements IteratorAggregate
```

Comment:
I think the `Aggregate` bit could probably be removed. Confusing from a reader PoV

Reply:
Its a shadow of the other one so should match for consistency

Reply:
:+1: